### PR TITLE
New guidelines for authors vs contributors

### PR DESCRIPTION
**Pull request by maggie-j-liu**
Added new guidelines for who should be added as an author and who should be added as a contributor. Also added a field in the module's frontmatter for contributors.

I will change some of the current authors to contributors once @darren-yao creates a list of the actual authors.

### CODE REVIEW

**Comment by sathu-beep** [comment]
I love this now people cant add 3 new lines under a code and write their name as one of the authors.

**Review by sathu-beep — APPROVED** [review]

**Comment by brad182** [comment]
i like the idea because then people who just make very minor changes will not be authors. 

however, a good chunk of internal solutions have no text at all and just have comments throughout the code (the official analysis is already linked). for example, someone who creates the internal solution and adds their python solution will get their name on as an author. When someone new adds their C++ implementation, based on these new rules, they will be a contributor even though they have done the same work as the original python guy who is listed as the author. just a concern that I have with these changes

**Comment by HarryWangATX** [comment]
> i like the idea because then people who just make very minor changes will not be authors.
> 
> however, a good chunk of internal solutions have no text at all and just have comments throughout the code (the official analysis is already linked). for example, someone who creates the internal solution and adds their python solution will get their name on as an author. When someone new adds their C++ implementation, based on these new rules, they will be a contributor even though they have done the same work as the original python guy who is listed as the author. just a concern that I have with these changes

mostly applies to modules

**Comment by maggie-j-liu** [comment]
These new guidelines are only for modules, internal solutions only have the authors field.

**Review by brad182 — APPROVED** [review]

**Comment by brad182** [comment]
> These new guidelines are only for modules, internal solutions only have the authors field.
> 
> Let me know if you have an idea on how to clarify that this only applies to modules and not internal solutions.

no it is clear I just didn't see the word module

File: content/1_General/Running_Code_Online.mdx [hunk 1]
```diff
@@ -1,7 +1,7 @@
 ---
 id: running-code-online
 title: Running Code Online
-author: Benjamin Qi, Hankai Zhang, Anthony Wang, Nathan Wang, Nathan Chen
```
Comment (SansPapyrus683) [inline content/1_General/Running_Code_Online.mdx:4]: what did they do to deserve this

**Review by SansPapyrus683 — COMMENTED** [review]
when were contributors a thing

File: content/1_General/Running_Code_Online.mdx (hunk 1 above)
Comment (maggie-j-liu) [reply content/1_General/Running_Code_Online.mdx:4]: i think stuff they wrote were moved to a different module

**Comment by HarryWangATX** [comment]
> when were contributors a thing

just been added

File: content/4_Gold/LIS.mdx [hunk 2]
```diff
@@ -1,9 +1,8 @@
 ---
 id: lis
 title: 'Longest Increasing Subsequence'
-author:
-  Michael Cao, Benjamin Qi, Andi Qu, Andrew Wang, Dong Liu, Siyong Huang,
-  Aryansh Shrivastava
+author: Michael Cao, Benjamin Qi, Andrew Wang
+contributors: Andi Qu, Dong Liu, Siyong Huang, Aryansh Shrivastava
```
Comment (dolphingarlic) [inline content/4_Gold/LIS.mdx:5]: Didn't I write the N log N LIS part?

Comment (dolphingarlic) [inline content/4_Gold/LIS.mdx:5]: Also, there's a "(By Aryansh Shrivastava)" lower down on this page, which we should probably remove

File: content/1_General/Working_MDX.mdx [hunk 3]
```diff
@@ -25,6 +26,8 @@ written in [YAML](https://yaml.org/). It stores the "metadata" for each module.
   based off this.
 - **Title**: _Required_. The title of the module. Ex: `Getting Started`
 - **Author**: _Required_. The author of the module. Ex: `Unknown`
+- **Contributors**: _Optional_. The people who contributed code and short
```
Comment (dolphingarlic) [inline content/1_General/Working_MDX.mdx:29]:
```suggestion
- **Contributors**: _Optional_. The people who contributed code and/or short
```

**Review by dolphingarlic — CHANGES_REQUESTED** [review]

File: content/4_Gold/Sliding.mdx [hunk 4]
```diff
@@ -1,7 +1,8 @@
 ---
 id: sliding
 title: 'Sliding Window'
-author: Darren Yao, Benjamin Qi, Andrew Wang
+author: Benjamin Qi
+contributors: Darren Yao, Andrew Wang, Andi Qu
```
Comment (dolphingarlic) [inline content/4_Gold/Sliding.mdx:5]: I think I only added problems to this module, so I probably shouldn't be here

File: content/4_Gold/LIS.mdx (hunk 2 above)
Comment (caoash) [reply content/4_Gold/LIS.mdx]: oh I must have missed something, my bad, didn't notice your changes in the history

**Review by caoash — APPROVED** [review]
Looks good to me.

**Pull request merged**